Classes/Utility/CacheUtility.php

```diff
@@ -68,11 +67,21 @@ public static function addPageCacheTagsByQuery(QueryInterface $query): void
             $cacheTags[] = 'tx_events2_domain_model_event';
         }
 
-        $GLOBALS['TSFE']->addCacheTags($cacheTags);
+        self::getTypoScriptFrontendController()->addCacheTags($cacheTags);
+    }
+
+    private static function getTypoScriptFrontendController(): TypoScriptFrontendController
+    {
+        return self::getTypo3Request()->getAttribute('frontend.controller');
+    }
+
+    private static function isFrontendRequest(): bool
+    {
+        return ApplicationType::fromRequest(self::getTypo3Request())->isFrontend();
     }
 
-    protected static function getEnvironmentService(): EnvironmentService
+    private static function getTypo3Request(): ServerRequestInterface
```

Comment:
make it public as it is `static`

Reply:
After renaming to CacheService this is not valid anymore